File: scr-edu/src/main/java/sorinpo/scr/edu/model/Info.java
package sorinpo.scr.edu.model;

import javax.persistence.Embedded;

import org.springframework.roo.addon.javabean.RooJavaBean;
import org.springframework.roo.addon.jpa.activerecord.RooJpaActiveRecord;
import org.springframework.roo.addon.json.RooJson;
import org.springframework.roo.addon.tostring.RooToString;

@RooJavaBean
@RooToString
@RooJson
@RooJpaActiveRecord(finders = { "findInfoesByUserIdAndYear" })
public class Info {
	
	public Info(long userId, int year){
		this.userId = userId;
		this.year = year;
	}
	
	public void initializeMonthlyNumbers(){
		beneficiariIndirecti = new MonthlyNumbers();
		voluntariImplicati = new MonthlyNumbers();
		intalniriGrupuriLucru = new MonthlyNumbers();
		participantiGrupuriLucru = new MonthlyNumbers();
		conferinteOrganizate = new MonthlyNumbers();
		participantiConferinte = new MonthlyNumbers();
		aparitiiPresa = new MonthlyNumbers();
	}
	
	private int year;
	Long userId;
	
	@Embedded
	private MonthlyNumbers beneficiariIndirecti;
	@Embedded
	private MonthlyNumbers voluntariImplicati;
	@Embedded
	private MonthlyNumbers intalniriGrupuriLucru;
	@Embedded
	private MonthlyNumbers participantiGrupuriLucru;
	@Embedded
	private MonthlyNumbers conferinteOrganizate;
	@Embedded
	private MonthlyNumbers participantiConferinte;
	@Embedded
	private MonthlyNumbers aparitiiPresa;
	
	private String links;

}

File: scr-edu/src/main/webapp/js/edu/model/Info.js
Ext.define('EDU.model.Info', {
    extend: 'Ext.data.Model',
    fields: [
       {name : 'id', type: 'auto', defaultValue: null }, 
       {name : 'version', type: 'auto', defaultValue: null },
       
       {name : 'year', type: 'int'},
       {name : 'userId', type: 'int'},
       
       {name : 'beneficiariIndirecti', type: 'auto', defaultValue: null },
       {name : 'voluntariImplicati', type: 'auto', defaultValue: null },
       {name : 'intalniriGrupuriLucru', type: 'auto', defaultValue: null },
       {name : 'participantiGrupuriLucru', type: 'auto', defaultValue: null },
       {name : 'conferinteOrganizate', type: 'auto', defaultValue: null },
       {name : 'participantiConferinte', type: 'auto', defaultValue: null },
       {name : 'aparitiiPresa', type: 'auto', defaultValue: null },
       
       {name : 'links', type: 'string', defaultValue: null }
    ],
    
    proxy: {
        type: 'rest',
        appendId: false,
        url: SERVER_ROOT + 'infos.json',
        getParams: Helpers.getParams({'username':'runas', 'year':'activeYear'}),
        reader: {
            type: 'json'
        }
    }
});